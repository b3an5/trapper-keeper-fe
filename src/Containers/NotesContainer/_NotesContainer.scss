.notes-container {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-gap: 3rem;
  margin: 3rem;
}