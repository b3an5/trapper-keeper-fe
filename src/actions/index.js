export const updateNotes = (notes) => ({
  type: 'UPDATE_NOTES',
  notes
})

export const toggleCompletedLi = (li) => ({
  type: 'TOGGLE_COMPLETED_LI',
  li
})

export const setCurrentNote = (note) => ({
  type: 'SET_CURRENT_NOTE',
  note
})