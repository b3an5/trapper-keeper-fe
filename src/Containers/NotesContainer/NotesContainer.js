import React, { Component } from 'react';
import PropTypes from 'prop-types'
import { connect } from 'react-redux';
import Note from '../Note/Note'
import { getNotes } from '../../utils/fetchCalls/getNotes'
import { updateNotes } from '../../actions/index'


export class NotesContainer extends Component  {

  componentDidMount = async () => {
    const notes = await getNotes();
    this.props.updateNotes(notes)  
  }

  render() {
    const notes = this.props.notes.map((note, i) => {
      return (
        <Note 
          title={note.title} 
          listItems={note.listItems} id={note.id}
          key={`00${i}_${note.title}`}/>
      )
    })
    return(
      <section className='notes-container'>
        {notes}
      </section>
    )
  }
}

NotesContainer.propTypes = {
  notes: PropTypes.arrayOf(PropTypes.object).isRequired,
  updateNotes: PropTypes.func.isRequired,
}


export const mapStateToProps = (state) => ({
  notes: state.notes || []
})

export const mapDispatchToProps = (dispatch) => ({
  updateNotes: (notes) => dispatch(updateNotes(notes))
})

export default connect(mapStateToProps, mapDispatchToProps)(NotesContainer)
