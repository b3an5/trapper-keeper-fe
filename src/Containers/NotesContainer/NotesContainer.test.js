import React from 'react';
import { shallow } from 'enzyme';
import { NotesContainer, mapStateToProps, mapDispatchToProps } from './NotesContainer';
import { updateNotes } from '../../actions'

describe('NotesContainer', () => {
  let wrapper;
  const mockNotes = [
    {
      id: 824, title: 'test', listItems: [
        { id: 248, text: 'test item' },
        { id: 2345, text: 'test item 2' }
      ]
    }
  ]; 

  const mockUpdateNotes=jest.fn();
  const mockGetNotes=jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <NotesContainer notes={ mockNotes } updateNotes={ mockUpdateNotes }/>
    )
  })

  it.skip('should invoke getNotes after ComponentDidMount', async () => {
    const spy = jest.spyOn(NotesContainer.prototype, 'getNotes')
    wrapper = mount(<NotesContainer {...props} />)

    await wrapper.instance().getNotes();

    expect(spy).toHaveBeenCalled()
  })

  it.skip('should invoke updateNotes after ComponentDidMount', () => {
    const spy = jest.spyOn(NotesContainer.prototype, 'updateNotes')
    wrapper = mount(<NotesContainer {...props} />)


  } )

  it.skip('should match component snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('mapStateToProps', () => {
    const mockState = { notes: [
      { id: 824, title: 'test', listItems: [
        { id: 248, text: 'test item'},
        { id: 2345, text: 'test item 2'}
      ]}
    ]};
    
    const expected = {
      notes: mockState.notes
    };

    const mappedProps = mapStateToProps(mockState);

    expect(mappedProps).toEqual(expected);
  });

  describe('mapDispatchToProps', () => {
    //setup
    const mockDispatch = jest.fn();
    const mockNotes = [
      { id: 824, title: 'test', listItems: [
        { id: 248, text: 'test item'},
        { id: 2345, text: 'test item 2'}
      ]}
    ];
    //execution
    const dispatchUpdateNotes = updateNotes(mockNotes);

    //assertion
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.updateNotes(mockNotes);
    expect(mockDispatch).toHaveBeenCalledWith(dispatchUpdateNotes);
  });
});
