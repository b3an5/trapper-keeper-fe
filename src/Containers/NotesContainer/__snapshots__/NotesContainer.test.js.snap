// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesContainer should match component snapshot 1`] = `undefined`;
